include/SDL3/SDL_stdinc.h

```diff
@@ -1230,13 +1475,104 @@ extern SDL_DECLSPEC int SDLCALL SDL_wcscasecmp(const wchar_t *str1, const wchar_
  */
 extern SDL_DECLSPEC int SDLCALL SDL_wcsncasecmp(const wchar_t *str1, const wchar_t *str2, size_t maxlen);
 
+/**
+ * Parse a `long` from a wide string.
+ *
+ * This function makes fewer guarantees than the C runtime `wcstol`:
+ *
+ * - Only the bases 10 and 16 are guaranteed to be supported. The behavior for other bases is unspecified.
+ * - It is unspecified what this function returns when the parsed integer does not fit inside a `long`.
+ *
+ * \param str The null-terminated wide string to read. Must not be NULL.
+ * \param endp If not NULL, the address of the first invalid wide character
+ *             (i.e. the next character after the parsed number) will be written to this pointer.
+ *             The address pointed to by `endp` must not overlap with `str`.
```

Comment:
I'm still not sure what this means. Can you give me an example where this constraint is violated?

Reply:
I ultimately ended up removing any mention of aliasing from these comments.

My understanding was that it's to guard against programs like these where the string data and the location the output address will be written to both overlap:
```c
#include <stdlib.h>
#include <stdio.h>

int main(void) {
    char buf[50] = "0123456789abcdef";

    char *s = buf;
    char **t = (char **)(&buf[8]);

    // any value written to '*t' will affect what 's' sees
    long long result = strtoll(s, t, 10);

    printf("result: %lld\n", result);
    printf("s: %p %s\n", s, s);
    printf("t: %p\n", *t);

    return 0;
}

// result: 1234567
// s: 0000001050BFFC70 01234567xⁿ┐P►
// t: 0000001050BFFC78
```
"Writing to `*t` must not what we can read from `s`!" is what the qualifier is trying to say, but this is already such a convoluted scenario it's unlikely to show up in actual code, plus it might already be UB, I'm not sure. Having it in the docs is probably only going to cause confusion. I googled around a bit and found a few discussions from mailing lists where even Gcc people seem to be confused about the meaning of `restrict` in `strtol` and similar functions.

Reply:
Ah, okay, that makes sense. I think removing the mention was the right call.